Bug:snappi: traffic_generation.py: dut_port_config is used as both list and dict.

### Is it platform specific

generic

### Importance or Severity

High

### Description of the bug

Seeing this problem when running snappi scripts in master:
```

            test_flow.metrics.enable = True
            test_flow.metrics.loss = True

            """ Set flow port config values """
            dut_port_config = base_flow_config["dut_port_config"]

            # Find the TX port config entry and append priority
            tx_peer_port = str(base_flow_config["tx_port_config"].peer_port)
>           for tx_port_dict in dut_port_config["Tx"]:
E           TypeError: list indices must be integers or slices, not str

base_flow_config = {'dut_port_config': [{'Ethernet136': []}, {'Ethernet128': []}], 'rx_mac': '90:88:55:46:06:f3', 'rx_port_config': <Snap...  prefix_len: 24, prefix_len_v6: None, type: SnappiPortType.IPInterface, peer_port: Ethernet128>, 'rx_port_id': 0, ...}
congested  = False
data_flow_config = {'flow_delay_sec': 4, 'flow_dur_sec': 304, 'flow_name': 'Test Flow', 'flow_pkt_count': None, ...}
dut_port_config = [{'Ethernet136': []}, {'Ethernet128': []}]
eth        = <snappi.snappi.FlowEthernet object at 0x7f38de923590>
flow_index = 0
ipv4       = <snappi.snappi.FlowIpv4 object at 0x7f38de923d10>
number_of_streams = 1
prio       = 4
prio_dscp_map = {0: [8], 1: [0, 1, 10, 11, 12, 13, ...], 2: [5], 3: [3], ...}
snappi_extra_params = <tests.common.snappi_tests.snappi_test_params.SnappiTestParams object at 0x7f388f4b8b50>
src_port   = 5008
test_flow  = <snappi.snappi.Flow object at 0x7f38df8a75e0>
test_flow_name = 'Test Flow Prio 4 Stream 0'
test_flow_name_dut_rx_port_map = {}
test_flow_name_dut_tx_port_map = {}
test_flow_prio_list = [4]
testbed_config = <snappi.snappi.Config object at 0x7f38da61d540>
tx_peer_port = 'Ethernet136'
udp        = <snappi.snappi.FlowUdp object at 0x7f38de923630>

common/snappi_tests/traffic_generation.py:248: TypeError
================================================================================================= warnings summary ==================================================================================================
```

The issue is that dut_port_config is treated as a list in some places, and dict in other places:

```
sonic@sonic-ucs-m4-3:~/rraghav/Ixia/100G-T2/master/sonic-test/sonic-mgmt/tests$ grep -rw dut_port_config  .
Binary file ./common/snappi_tests/__pycache__/traffic_generation.cpython-38.pyc matches
./common/snappi_tests/traffic_generation.py:        base_flow_config (dict): base flow configuration containing dut_port_config, tx_mac,
./common/snappi_tests/traffic_generation.py:                dut_port_config (dict): a dictionary with "Tx" and "Rx" keys, each containing a list of dictionaries
./common/snappi_tests/traffic_generation.py:    dut_port_config = {"Tx": [], "Rx": []}
./common/snappi_tests/traffic_generation.py:        dut_port_config["Tx"].append(tx_dict)
./common/snappi_tests/traffic_generation.py:        dut_port_config["Rx"].append(rx_dict)
./common/snappi_tests/traffic_generation.py:    base_flow_config["dut_port_config"] = dut_port_config
./common/snappi_tests/traffic_generation.py:        dut_port_config = base_flow_config["dut_port_config"]
./common/snappi_tests/traffic_generation.py:        for tx_port_dict in dut_port_config["Tx"]:
./common/snappi_tests/traffic_generation.py:        for rx_port_dict in dut_port_config["Rx"]:
./common/snappi_tests/traffic_generation.py:        base_flow_config["dut_port_config"] = dut_port_config
./common/snappi_tests/traffic_generation.py:    switch_tx_lossless_prios = sum(base_flow_config["dut_port_config"]["Tx"][0].values(), [])
./common/snappi_tests/traffic_generation.py:    dut_port_config = snappi_extra_params.base_flow_config["dut_port_config"]
./common/snappi_tests/traffic_generation.py:    pytest_assert(dut_port_config is not None, "Flow port config is not provided")
./common/snappi_tests/traffic_generation.py:        for peer_port, prios in dut_port_config["Tx"][0].items():
./common/snappi_tests/traffic_generation.py:        for peer_port, prios in dut_port_config["Tx"][0].items():
./common/snappi_tests/traffic_generation.py:    dut_port_config = snappi_extra_params.base_flow_config["dut_port_config"]
./common/snappi_tests/traffic_generation.py:    pytest_assert(dut_port_config is not None, 'Flow port config is not provided')
./common/snappi_tests/traffic_generation.py:    for peer_port, prios in dut_port_config["Rx"][0].items():  # PFC pause frames received on DUT's egress port
./common/snappi_tests/traffic_generation.py:    for peer_port, prios in dut_port_config["Tx"][0].items():  # PFC pause frames sent by DUT's ingress port to TGEN
./common/snappi_tests/traffic_generation.py:    dut_port_config = snappi_extra_params.base_flow_config["dut_port_config"]
./common/snappi_tests/traffic_generation.py:    pytest_assert(dut_port_config is not None, 'Flow port config is not provided')
./common/snappi_tests/traffic_generation.py:    for peer_port, _ in dut_port_config["Rx"][0].items():
./common/snappi_tests/traffic_generation.py:    dut_port_config = snappi_extra_params.base_flow_config["dut_port_config"]
./common/snappi_tests/traffic_generation.py:    pytest_assert(dut_port_config is not None, 'Flow port config is not provided')
./common/snappi_tests/traffic_generation.py:    for peer_port, _ in dut_port_config["Tx"][0].items():
./common/snappi_tests/traffic_generation.py:    dut_port_config = snappi_extra_params.base_flow_config["dut_port_config"]
./common/snappi_tests/traffic_generation.py:    pytest_assert(dut_port_config is not None, 'Flow port config is not provided')
./common/snappi_tests/traffic_generation.py:        for peer_port, prios in dut_port_config["Rx"][0].items():
./common/snappi_tests/traffic_generation.py:    # assign base_flow_config[0]["dut_port_config"] to dut_port_config.
./common/snappi_tests/traffic_generation.py:        dut_port_config = snappi_extra_params.base_flow_config["dut_port_config"]
./common/snappi_tests/traffic_generation.py:        dut_port_config = snappi_extra_params.base_flow_config_list[0]["dut_port_config"]
./common/snappi_tests/traffic_generation.py:    pytest_assert(dut_port_config is not None, 'Flow port config is not provided')
./common/snappi_tests/traffic_generation.py:        for peer_port, prios in dut_port_config["Rx"][0].items():
./common/snappi_tests/traffic_generation.py:    rx_port_config_values = snappi_extra_params.base_flow_config_list[0]["dut_port_config"]["Rx"][0].values()
./common/snappi_tests/traffic_generation.py:        base_flow_config (dict): base flow configuration containing dut_port_config, tx_mac,
./common/snappi_tests/traffic_generation.py:                dut_port_config (list): a list of two dictionaries of tx and rx ports on the peer (switch) side,
./common/snappi_tests/traffic_generation.py:    # Instantiate peer ports in dut_port_config
./common/snappi_tests/traffic_generation.py:    dut_port_config = []
./common/snappi_tests/traffic_generation.py:    dut_port_config.append(tx_dict)
./common/snappi_tests/traffic_generation.py:    dut_port_config.append(rx_dict)
./common/snappi_tests/traffic_generation.py:    base_flow_config["dut_port_config"] = dut_port_config
./snappi_tests/pfc/files/valid_src_mac_pfc_frame_helper.py:dut_port_config = []
./snappi_tests/pfc/files/helper.py:dut_port_config = []
./snappi_tests/pfc/files/mixed_speed_multidut_helper.py:dut_port_config = []
./snappi_tests/pfc/files/pfc_congestion_helper.py:dut_port_config = []
./snappi_tests/pfcwd/files/pfcwd_actions_helper.py:dut_port_config = []
sonic@sonic-ucs-m4-3:~/rraghav/Ixia/100G-T2/master/sonic-test/sonic-mgmt/tests$ 
```

@selldinesh , @kamalsahu0001 :  Pls fix this. thanks.

Running script:
```
================================================================================================= warnings summary ==================================================================================================
../../usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236
  /usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
    "class": algorithms.Blowfish,

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
------------------------------------------------------------ generated xml file: /run_logs/ixia/macsec-1/2026-01-30-02-03-33/tr_2026-01-30-02-03-33.xml -------------------------------------------------------------
INFO:root:Can not get Allure report URL. Please check logs
============================================================================================== short test summary info ==============================================================================================
SKIPPED [9] common/helpers/assertions.py:19: Need Minimum of 2 ports defined in ansible/files/*links.csv file
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_diff_dist[multidut_port_info0-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_diff_dist[multidut_port_info1-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_diff_dist[multidut_port_info2-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_uni_dist[multidut_port_info0-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_uni_dist[multidut_port_info1-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_uni_dist[multidut_port_info2-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_single_lossless_prio[multidut_port_info0-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_single_lossless_prio[multidut_port_info1-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_single_lossless_prio[multidut_port_info2-port_map0] - TypeError: list indices must be integers or slices, not str
=============================================================================== 9 failed, 9 skipped, 1 warning in 2724.69s (0:45:24) ================================================================================
sonic@snappi-sonic-mgmt-100g-t2-master:/data/tests$ 
```

### Steps to Reproduce


run the script:
```
================================================================================================= warnings summary ==================================================================================================
../../usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236
  /usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
    "class": algorithms.Blowfish,

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
------------------------------------------------------------ generated xml file: /run_logs/ixia/macsec-1/2026-01-30-02-03-33/tr_2026-01-30-02-03-33.xml -------------------------------------------------------------
INFO:root:Can not get Allure report URL. Please check logs
============================================================================================== short test summary info ==============================================================================================
SKIPPED [9] common/helpers/assertions.py:19: Need Minimum of 2 ports defined in ansible/files/*links.csv file
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_diff_dist[multidut_port_info0-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_diff_dist[multidut_port_info1-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_diff_dist[multidut_port_info2-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_uni_dist[multidut_port_info0-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_uni_dist[multidut_port_info1-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_multiple_prio_uni_dist[multidut_port_info2-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_single_lossless_prio[multidut_port_info0-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_single_lossless_prio[multidut_port_info1-port_map0] - TypeError: list indices must be integers or slices, not str
FAILED snappi_tests/pfc/test_pfc_no_congestion_throughput.py::test_single_lossless_prio[multidut_port_info2-port_map0] - TypeError: list indices must be integers or slices, not str
=============================================================================== 9 failed, 9 skipped, 1 warning in 2724.69s (0:45:24) ================================================================================
sonic@snappi-sonic-mgmt-100g-t2-master:/data/tests$ 
```

### Actual Behavior and Expected Behavior

It should pass.

### Relevant log output

```shell

```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/common/snappi_tests/traffic_generation.py
+++ b/tests/common/snappi_tests/traffic_generation.py
@@ -2011,9 +2011,10 @@ def multi_base_traffic_config(testbed_config,
                 rx_mac (str): MAC address of ixia RX port ex. '00:00:fa:ce:fa:ce'
                 tx_port_name (str): name of ixia TX port ex. 'Port 1'
                 rx_port_name (str): name of ixia RX port ex. 'Port 2'
-                dut_port_config (list): a list of two dictionaries of tx and rx ports on the peer (switch) side,
-                                        and the associated test priorities
-                                        ex. [{'Ethernet4':[3, 4]}, {'Ethernet8':[3, 4]}]
+                dut_port_config (dict): a dictionary with "Tx" and "Rx" keys, each containing a list of dictionaries
+                                        of tx and rx ports on the peer (switch) side, and the associated test
+                                        priorities ex. {"Tx": [{'Ethernet4':[3, 4]}],
+                                        "Rx": [{'Ethernet8':[3, 4]}]}
                 test_flow_name_dut_rx_port_map (dict): Mapping of test flow name to DUT RX port(s)
                                                   ex. {'flow1': [Ethernet4, Ethernet8]}
                 test_flow_name_dut_tx_port_map (dict): Mapping of test flow name to DUT TX port(s)
@@ -2029,11 +2030,11 @@ def multi_base_traffic_config(testbed_config,
     base_flow_config["rx_port_config"] = rx_port_config
 
     # Instantiate peer ports in dut_port_config
-    dut_port_config = []
+    dut_port_config = {"Tx": [], "Rx": []}
     tx_dict = {str(tx_port_config.peer_port): []}
     rx_dict = {str(rx_port_config.peer_port): []}
-    dut_port_config.append(tx_dict)
-    dut_port_config.append(rx_dict)
+    dut_port_config["Tx"].append(tx_dict)
+    dut_port_config["Rx"].append(rx_dict)
     base_flow_config["dut_port_config"] = dut_port_config
 
     base_flow_config["tx_mac"] = tx_port_config.mac

--- a/tests/common/snappi_tests/traffic_generation.py
+++ b/tests/common/snappi_tests/traffic_generation.py
@@ -2019,6 +2019,12 @@ def multi_base_traffic_config(testbed_config,
                                                   ex. {'flow1': [Ethernet4, Ethernet8]}
                 test_flow_name_dut_tx_port_map (dict): Mapping of test flow name to DUT TX port(s)
                                                   ex. {'flow1': [Ethernet4, Ethernet8]}
+
+    Note:
+        This helper currently creates a single port-mapping dictionary under each
+        of the "Tx" and "Rx" lists. In contrast, setup_base_traffic_config() may
+        produce multiple port-mapping dictionaries per direction. Both helpers
+        follow the same {"Tx", "Rx"} contract.
     """
     base_flow_config = {}
     base_flow_config["rx_port_id"] = rx_port_id

